### ISSUE TITLE
CSV Import Tool Fails On Reverse Proxy SSL (Potential fix)

### ISSUE THREAD

**Issue by LukeStaszewski** [firefly-iii/firefly-iii#4302]
**Bug description**
I am running the latest version of the CSV Importer script but I'm having some issues getting it to work. I think I know the solution to the bug but I am struggling to implement it because I lack the Laravel knowledge.

**Steps to reproduce**

Install Firefly CSV Importer using Yacht (or Portainer, or just Docker) and then setup a Nginx Reverse Proxy to a subdomain (like fireflyimporter.yourdomain.com) and then get an SSL for the subdomain via Lets Encrypt.

The page will load in non-https (but the import doesn't work then, i think it needs SSL to work if I remember reading the docs correctly, apologies if not) and the SSL subdomain (setup via NGinx) will load but it does not load the non-https content in the bootstrap script.

The problem files are: 

/css/bootstrap.min.css?version=2.3.4
/fa/css/all.min.css
/js/jquery-3.4.1.min.js?version=2.3.4
/js/popper.min.js?version=2.3.4

These all load as non-https even when the domain is https (Maybe just when setup via reverse proxy? not sure if that makes a difference)
But I located these files in the Github and I was going to attempt to change them to https:// links and see if that fixes everything and if so, submit a pull request.

But I found the correct code in the files (I believe so... not very good at laravel) but I am not sure what to do to change it. It loads them like this:

<script src="{{ asset('js/jquery-3.4.1.min.js') }}?version={{ version }}"></script>
<script src="{{ asset('js/popper.min.js') }}?version={{ version }}"></script>
<script src="{{ asset('js/bootstrap.min.js') }}?version={{ version }}"></script>
{% block scripts %}{% endblock %}

But I'm not sure where to edit the actual variables and change the links from http to https.

I have searched for an answer for this in Github and on Google but have not had any luck in finding an answer. I apologize if I've missed something obvious, but I tried my best to fix this bug myself and contribute to the project. Hopefully my efforts are helpful and one of you smart folk can help me finish the job. I really appreciate any help.

Please let me know if I've missed any important information. I tried my best to explain things as best I could, apologies if I have not done so correctly.

Kind regards,
Luke

**Comment by JC5** [comment]
You need something like this in your reverse proxy config.

```
server {
 
    location / {
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header Host $host;

        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Forwarded-Proto $scheme;

    }
}
```

**Comment by LukeStaszewski** [comment]
Hi @JC5 thanks so much for replying so quickly. Thanks a ton!

I tried to implement that fix but it hasn't seemed to have changed anything. It might need some time for cache to clear or something? I'm off to bed now but just thought I'd show you my config in case I did anything wrong. I'll update again in the morning if it hasn't fixed itself by then
![image](https://user-images.githubusercontent.com/60547309/105824305-12db9880-5fb6-11eb-8679-ec0f75b85d91.png)


Thanks again for the help so far :D
